src/core/server/integration_tests/http/versioned_router.test.ts

```diff
@@ -340,4 +342,21 @@ describe('Routing versioned requests', () => {
         .then(({ body }) => body.v)
     ).resolves.toEqual('oldest');
   });
+
+  it('captures the error if handler throws', async () => {
+    const error = new Error(`some error`);
+
+    router.versioned
+      .get({ path: '/my-path', access: 'internal' })
+      .addVersion({ validate: false, version: '1' }, async (ctx, req, res) => {
+        throw error;
+      });
+
+    await server.start();
+
+    await supertest.get('/my-path').set('Elastic-Api-Version', '1').expect(500);
+
+    expect(captureErrorMock).toHaveBeenCalledTimes(1);
```

Comment:
NIT it'd be interesting to reset the `captureErrorMock` in case we add more tests that use it.